File: app/src/main/java/com/iceteaviet/fastfoodfinder/data/local/user/model/UserEntity.java
package com.iceteaviet.fastfoodfinder.data.local.user.model;

import com.iceteaviet.fastfoodfinder.data.remote.user.model.User;
import com.iceteaviet.fastfoodfinder.data.remote.user.model.UserStoreList;

import java.util.List;

import io.realm.RealmList;
import io.realm.RealmObject;

/**
 * Created by Genius Doan on 2018.
 */
public class UserEntity extends RealmObject {

    private String name;
    private String email;
    private String uid;
    private String photoUrl;
    private RealmList<UserStoreListEntity> userStoreListEntities;

    public void map(User user) {
        name = user.getName();
        email = user.getEmail();
        uid = user.getUid();
        photoUrl = user.getPhotoUrl();
        this.userStoreListEntities = new RealmList<>();
        List<UserStoreList> userStoreLists = user.getUserStoreLists();
        for (int i = 0; i < userStoreLists.size(); i++) {
            this.userStoreListEntities.add(new UserStoreListEntity(userStoreLists.get(i)));
        }
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getUid() {
        return uid;
    }

    public void setUid(String uid) {
        this.uid = uid;
    }

    public String getPhotoUrl() {
        return photoUrl;
    }

    public void setPhotoUrl(String photoUrl) {
        this.photoUrl = photoUrl;
    }

    public RealmList<UserStoreListEntity> getUserStoreLists() {
        return userStoreListEntities;
    }

    public void setUserStoreLists(RealmList<UserStoreListEntity> userStoreLists) {
        this.userStoreListEntities = userStoreLists;
    }
}

File: app/src/main/java/com/iceteaviet/fastfoodfinder/data/local/user/model/UserEntity.kt
package com.iceteaviet.fastfoodfinder.data.local.user.model

import com.iceteaviet.fastfoodfinder.data.remote.user.model.User
import io.realm.RealmList
import io.realm.RealmObject

/**
 * Created by Genius Doan on 2018.
 */
open class UserEntity : RealmObject() {
    var name: String = ""
    var email: String = ""
    private var uid: String = ""
    var photoUrl: String = ""
    var userStoreLists: RealmList<UserStoreListEntity> = RealmList()

    fun getUid(): String {
        return uid
    }

    fun map(user: User) {
        name = user.name
        email = user.email
        uid = user.getUid()
        photoUrl = user.photoUrl
        this.userStoreLists = RealmList()
        val userStoreLists = user.getUserStoreLists()
        for (i in userStoreLists.indices) {
            this.userStoreLists.add(UserStoreListEntity(userStoreLists.get(i)))
        }
    }
}
